DSharpPlus/Clients/DiscordClient.WebSocket.cs

```diff
@@ -171,7 +171,9 @@ async Task SocketOnDisconnect(IWebSocketClient sender, SocketCloseEventArgs e)
                 this.Logger.LogDebug(LoggerEvents.ConnectionClose, "Connection closed ({0}, '{1}')", e.CloseCode, e.CloseMessage);
                 await this._socketClosed.InvokeAsync(this, e).ConfigureAwait(false);
 
-                if (this.Configuration.AutoReconnect)
+
+                //CloseCode 4014 is disallowed Intent and CloseCode 4013 is invalid Intent.  We Should not continuously reconnect per issue 682
+                if (this.Configuration.AutoReconnect && e.CloseCode != 4014 && e.CloseCode != 4013)
```

Comment:
Should not reconnect on any 4xxx other than 4000 since we use that one for """clean""" close.

Reply:
Change applied